### PR TITLE
feat: register spends in the network

### LINKED ISSUE #117: Register spends in the network.

### PR DESCRIPTION
**Pull request by oetyng**
Resolves #117.

- This is the final step in making a transfer valid in the network.
- Fees are paid to nodes.

***

**NB1**: Some additional validation of responses is needed to make sure we
error if not enough nodes could handle the request.

**NB2**: Nodes still need to store the rewards in their wallet, TBD.

**NB3**: There are still some code reuse work to be done between
transfer online and offline files.

### CODE REVIEW

File: safenode/src/protocol/transfers/online.rs [hunk 1]
```diff
@@ -350,3 +400,38 @@ async fn get_fees(dbc_id: DbcId, client: &Client) -> Result<BTreeMap<NodeId, Req
...
+/// so that a node can find and verify the fee paid to them, from
+/// within a request to spend a dbc.
+///
+/// Within all the outputs, will be each fee payment to respective node.
+/// We will encrypt the derivation index to each node public addresss, and
+/// the amount to the derived key corresponding to the dbc id of each node fee payment.
+#[allow(clippy::result_large_err)]
+fn fee_ciphers(
+    outputs: &BTreeMap<DbcId, RevealedAmount>,
+    fee_cipher_params: &BTreeMap<NodeId, (RequiredFee, DbcIdSource)>,
```
Comment (grumbach) [inline safenode/src/protocol/transfers/online.rs:414]: Param feels weird for a function parameter I'd suggest renaming that to `fees_per_node`

**Review by grumbach — APPROVED** [review]
LGTM

File: safenode/src/protocol/transfers/online.rs (hunk 1 above)
Comment (oetyng) [reply safenode/src/protocol/transfers/online.rs:414]: Yeah, good point. That name was a bit temporary.

Comment (oetyng) [reply safenode/src/protocol/transfers/online.rs:414]: I wanted something that conveys that these are to be used for creating the ciphers.

Hmm... But I will think of something.

**Pull request merged**